bin/proving-service/README.md

```diff
@@ -28,64 +28,49 @@ This will spawn a worker using the hosts and ports defined in the command option
-health_check_interval_secs = 1
-# Host of the metrics server
-prometheus_host = "127.0.0.1"
-# Port of the metrics server
-prometheus_port = 6192
+miden-proving-service start-proxy [worker1] [worker2] ... [workerN]
 ```
 
-Then, to start the proxy service, you will need to run:
+For example:
 
 ```bash
-miden-proving-service start-proxy [worker1] [worker2] ... [workerN]
+miden-proving-service start-proxy 0.0.0.0:8084 0.0.0.0:8085
 ```
 
 This command will start the proxy using the workers passed as arguments. The workers should be in the format `host:port`. If no workers are passed, the proxy will start without any workers and will not be able to handle any requests until one is added through the `miden-proving-service add-worker` command.
 
-At the moment, when a worker added to the proxy stops working and can not connect to it for a request, the connection is marked as retriable meaning that the proxy will try reaching another worker. The number of retries is configurable via the `max_retries_per_request` value in the configuration file.
+You can customize the proxy service by setting environment variables. Possible customizations can be found by running `miden-proving-service start-proxy --help`.
+
+An example `.env` file is provided in the crate's root directory. To use the variables from a file in any Unix-like operating system, you can run `source <your-file>`.
+
+At the moment, when a worker added to the proxy stops working and can not connect to it for a request, the connection is marked as retriable meaning that the proxy will try reaching another worker. The number of retries is configurable via the `MPS_MAX_RETRIES_PER_REQUEST` environmental variable.
 
 ## Updating workers on a running proxy
 
 To update the workers on a running proxy, two commands are provided: `add-worker` and `remove-worker`. These commands will update the workers on the proxy and will not require a restart. To use these commands, you will need to run:
 
 ```bash
-miden-proving-service add-worker [worker1] [worker2] ... [workerN]
-miden-proving-service remove-worker [worker1] [worker2] ... [workerN]
+miden-proving-service add-worker --proxy-host <proxy-host> --proxy-update-workers-port <proxy-update-workers-port> [worker1] [worker2] ... [workerN]
+miden-proving-service remove-worker --proxy-host <proxy-host> --proxy-update-workers-port <proxy-update-workers-port> [worker1] [worker2] ... [workerN]
 ```
 For example:
 
 ```bash
 # To add 0.0.0.0:8085 and 200.58.70.4:50051 to the workers list:
-miden-proving-service add-workers 0.0.0.0:8085 200.58.70.4:50051
+miden-proving-service add-workers --proxy-host 0.0.0.0 --proxy-update-workers-port 8083 0.0.0.0:8085 200.58.70.4:50051
 # To remove 158.12.12.3:8080 and 122.122.6.6:50051 from the workers list:
-miden-proving-service remove-workers 158.12.12.3:8080 122.122.6.6:50051
+miden-proving-service remove-workers --proxy-host 0.0.0.0 --proxy-update-workers-port 8083 158.12.12.3:8080 122.122.6.6:50051
```

Comment:
Not for this PR, but I'm wondering if we should combine host and port into a single variable (maybe using a `Url` type as we did in the node). Let's create an issue for this.

Reply:
I created https://github.com/0xPolygonMiden/miden-base/issues/1130